btx/processing/merge.py

```diff
@@ -0,0 +1,262 @@
+    outfile : str
+        path for saving plot, optional
+        
+    Returns
+    -------
+    fom : str
+        figure of merit
+    stat : float
+        overall figure of merit value
+    """  
+    shells = np.loadtxt(shells_file, skiprows=1)
+    with open(shells_file) as f:
+        lines = f.read() 
+        header = lines.split('\n')[0]
+    fom = header.split()[2]
+    
+    if len(shells.shape)==1:
+        if outfile is not None:
+            print("The .dat file was generated with only one shell so cannot be plotted.")
+        return fom, shells[1]
+    
+    if len(shells.shape) > 1:
+        f, ax1 = plt.subplots(figsize=(6,4))
+
+        ax1.plot(shells[:,0], shells[:,1], c='black')
+        ax1.scatter(shells[:,0], shells[:,1], c='black')
+
+        ticks = ax1.get_xticks()
+        ax1.set_xticklabels(["{0:0.2f}".format(i) for i in 10/ticks])
+        ax1.tick_params(axis='both', which='major', labelsize=12)
+        ax1.set_xlabel("Resolution ($\mathrm{\AA}$)", fontsize=14)
+        ax1.set_ylabel(f"{fom}", fontsize=14)
+
+        if outfile is not None:
+            f.savefig(outfile, dpi=300, bbox_inches='tight')
+
+def parse_input():
+    """
+    Parse command line input.
+    """
+    parser = argparse.ArgumentParser()
+    # global arguments
+    parser.add_argument('-i', '--input_stream', required=True, type=str, help='Input stream file')
+    parser.add_argument('--symmetry', required=True, type=str, help='Point group symmetry')
+    parser.add_argument('--taskdir', required=True, type=str, help='Base directory for indexing results')
+    parser.add_argument('--cell', required=True, type=str, help='File containing unit cell information (.pdb or .cell)')
+    # arguments specific to partialator
+    parser.add_argument('--model', default='unity', choices=['unity','xsphere'], type=str, help='Partiality model')
+    parser.add_argument('--iterations', default=1, type=int, help='Number of cycles of scaling and post-refinement to perform')
+    parser.add_argument('--min_res', required=False, type=float, help='Minimum resolution for crystal to be merged')
+    parser.add_argument('--push_res', required=False, type=float, help='Maximum resolution beyond min_res for reflection to be merged')
+    # arguments related to computing figures of merit and reporting
+    parser.add_argument('--foms', required=False, type=str, nargs='+', help='Figures of merit to calculate')
+    parser.add_argument('--nshells', required=False, type=float, default=10, help='Number of resolution shells for computing figures of merit')
+    parser.add_argument('--highres', required=False, type=float, help='High resolution limit for computing figures of merit') 
+    parser.add_argument('--report', help='Report indexing results to summary file and elog', action='store_true')
+    parser.add_argument('--update_url', help='URL for communicating with elog', required=False, type=str)
+    
+    return parser.parse_args()
```

Comment:
When we'll start testing, a good test will be to check the consistency between this argument list and what is expected by __init__(), etc.

Reply:
To be addressed in a new issue.